devito/function.py

```diff
@@ -744,6 +748,24 @@ def dt2(self):
 
         return self.diff(_t, _t).as_finite_difference(indt)
 
+    def argument_values(self, alias=None, **kwargs):
+        """
+        Returns a map of argument values after evaluating user input.
+
+        :param kwargs: Dictionary of user-provided argument overrides.
+        :param alias: (Optional) name under which to store values.
+        """
+        saves = {True: 'int', False: 'None'}
+        # Check if data has the right dimension
+        if self.name in kwargs:
+            new = kwargs.get(self.name)
+            if isinstance(new, TimeFunction) and new.save != self.save:
+                raise TypeError("Incorrect value encounterd, save should be %s" %
+                                saves[self.save])
```

Comment:
``save should be %s" % type(self.save))``

ie, no need for ``saves``

Reply:
`self.save` is True or False, type(self.save) would return `<class 'bool'>` both cases instead of `int, None`

Reply:
done